### DISCUSSION TITLE
How "boolean props default problem" could be resolved?

### DISCUSSION THREAD

**Discussion by extrem7** [vuejs/language-tools#1243] [General]
Hello there!

Currently, vue-tsc forces users to bind boolean prop's default value `:props="false"`. You may mark a prop as optional, but it will cause potential problems with `toRefs` and passing this prop to nested components that expect boolean (not boolean|undefined).

As @johnsoncodehk [noticed](https://github.com/johnsoncodehk/volar/issues/1224#issuecomment-1107393622) using default value (withDefaults for script setup) will solve this issue, but at this point [vue/no-boolean-default](https://eslint.vuejs.org/rules/no-boolean-default.html) enters the game.

In my mind, there is no sense in current Volar's behavior since this problem was disscussed in [vuejs/core](https://github.com/vuejs/core/issues/3288) and resolved by [yyx990803](https://github.com/vuejs/core/commit/4fe4de0a49ffc2461b0394e74674af38ff5e2a20) year ago. So boolean props should be false by default without any additional `?:`, withDefaults, etc.

**Comment by johnsoncodehk** [comment]
See #1224.
